src/dd/util.rs

```diff
@@ -0,0 +1,180 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::sync::{Arc, Mutex};
+
+use super::{chan};
+
+use super::defs::{Config};
+
+fn dd_out(buf: &mut Vec<u8>, pos: usize, num: usize, out: &mut ::std::io::Write) -> Option<usize> {
+    match out.write(&buf[pos..pos + num]) {
+        Ok(num) if num != 0 => Some(num),
+        Ok(0) | _ => None,
+    }
+}
+
+fn copy(cf: Arc<Mutex<Config>>) {
+    use std::{cmp, io};
+    use std::fs::File;
+    let (is_inp_std, is_out_std, i_name, o_name, count, inp_buf_size, out_buf_size) = {
+        let conf = cf.lock().unwrap();
+        (
+            conf.in_.process_std_in_out,
+            conf.out.process_std_in_out,
+            conf.in_.name.clone(),
+            conf.out.name.clone(),
+            conf.copy_count,
+            conf.in_.buf_size,
+            conf.out.buf_size,
+        )
+    };
+    let mut sin;
+    let mut sout;
+    let mut i_f;
+    let mut o_f;
+    let in_: &mut io::Read = if is_inp_std {
+        sin = io::stdin();
+        &mut sin
+    } else {
+        let inp = File::open(&i_name);
+        if inp.is_err() {
```

Comment:
This would probably be better as a `match`.  Same for the one down below as well.

Reply:
Ok.